src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -674,9 +696,14 @@ export class NotebookModel extends Disposable implements INotebookModel {
 			this._cells.push(cell);
 			index = undefined;
 		}
-		// Set newly created cell as active cell
-		this.updateActiveCell(cell);
-		cell.isEditMode = true;
+		if (addToUndoStack) {
+			this.undoService.pushElement(new AddCellEdit(this, cell, index));
+
+			cell.isEditMode = true;
```

Comment:
Why is this logic in the addToUndoStack block only? This stuff shouldn't be tied to whether we add it to the history or not